pkg/utils/utils.go

```diff
@@ -244,6 +244,35 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 				if iface.Mtu > 0 {
 					mtu = iface.Mtu
 				}
+
+				// validate the device exist only with the default driver
+				// TODO: remove this work around after the BZ is close
+				// https://bugzilla.redhat.com/show_bug.cgi?id=1875338
+				b := backoff.NewConstantBackOff(1 * time.Second)
+				err = backoff.Retry(func() error {
+					_, err := dputils.GetNetNames(addr)
```

Comment:
When the VFs first get created, `setVfsAdminMac` (line 205) sets the initial VF admin mac address, in the same function, it checks whether VF is ready by getting vflink via its interface name. If VF doesn't appear as a kernel interface, it should fail earlier in line 205?

Reply:
Looks like it passes the line 205 and then the file does not exist anymore.
here is another log

```
I0921 15:42:34.739112 2991726 generic_plugin.go:105] generic-plugin Apply(): desiredState={1003117 [{0000:12:00.0 5 0 ens1f0 eth [{testresource netdevice 0-4 test-policyq9ll6}]}]}
I0921 15:42:34.739201 2991726 utils.go:154] needUpdate(): NumVfs needs update desired=5, current=0
I0921 15:42:34.739216 2991726 utils.go:187] configSriovDevice(): config interface 0000:12:00.0 with &{0000:12:00.0 5 0 ens1f0 eth [{testresource netdevice 0-4 test-policyq9ll6}]}
I0921 15:42:34.739230 2991726 utils.go:264] setSriovNumVfs(): set NumVfs for device 0000:12:00.0
I0921 15:42:35.262028 2991726 utils.go:467] setVfsAdminMac(): device 0000:12:00.0
I0921 15:42:35.262772 2991726 utils.go:449] vfIsReady(): VF device 0000:12:02.0
E0921 15:42:35.262834 2991726 utils.go:456] vfIsReady(): unable to get VF link for device 0000:12:02.0, "Link not found"
I0921 15:42:36.263006 2991726 utils.go:309] tryGetInterfaceName(): name is ens1f0v0
I0921 15:42:36.372764 2991726 driver.go:19] Unbind(): unbind device 0000:12:02.0 driver
I0921 15:42:36.373074 2991726 driver.go:112] hasDriver(): device 0000:12:02.0 driver is iavf
I0921 15:42:36.540667 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.0 to default driver
I0921 15:42:36.540710 2991726 driver.go:109] hasDriver(): device 0000:12:02.0 driver is empty
I0921 15:42:36.542501 2991726 utils.go:449] vfIsReady(): VF device 0000:12:02.1
I0921 15:42:36.542547 2991726 utils.go:309] tryGetInterfaceName(): name is ens1f0v1
I0921 15:42:36.773683 2991726 driver.go:19] Unbind(): unbind device 0000:12:02.1 driver
I0921 15:42:36.773816 2991726 driver.go:112] hasDriver(): device 0000:12:02.1 driver is iavf
I0921 15:42:36.946663 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.1 to default driver
I0921 15:42:36.946702 2991726 driver.go:109] hasDriver(): device 0000:12:02.1 driver is empty
I0921 15:42:37.022179 2991726 utils.go:449] vfIsReady(): VF device 0000:12:02.2
I0921 15:42:37.022221 2991726 utils.go:309] tryGetInterfaceName(): name is ens1f0v2
I0921 15:42:37.184238 2991726 driver.go:19] Unbind(): unbind device 0000:12:02.2 driver
I0921 15:42:37.184263 2991726 driver.go:112] hasDriver(): device 0000:12:02.2 driver is iavf
I0921 15:42:37.329525 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.2 to default driver
I0921 15:42:37.329561 2991726 driver.go:109] hasDriver(): device 0000:12:02.2 driver is empty
I0921 15:42:37.345525 2991726 utils.go:449] vfIsReady(): VF device 0000:12:02.3
I0921 15:42:37.345580 2991726 utils.go:309] tryGetInterfaceName(): name is ens1f0v3
I0921 15:42:37.569585 2991726 driver.go:19] Unbind(): unbind device 0000:12:02.3 driver
I0921 15:42:37.569610 2991726 driver.go:112] hasDriver(): device 0000:12:02.3 driver is iavf
I0921 15:42:37.712516 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.3 to default driver
I0921 15:42:37.712550 2991726 driver.go:109] hasDriver(): device 0000:12:02.3 driver is empty
I0921 15:42:37.728871 2991726 utils.go:449] vfIsReady(): VF device 0000:12:02.4
I0921 15:42:37.728909 2991726 utils.go:309] tryGetInterfaceName(): name is ens1f0v4
I0921 15:42:37.937258 2991726 driver.go:19] Unbind(): unbind device 0000:12:02.4 driver
I0921 15:42:37.937283 2991726 driver.go:112] hasDriver(): device 0000:12:02.4 driver is iavf
I0921 15:42:38.079510 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.4 to default driver
I0921 15:42:38.079549 2991726 driver.go:109] hasDriver(): device 0000:12:02.4 driver is empty
I0921 15:42:38.095910 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.0 to default driver
I0921 15:42:38.095926 2991726 driver.go:112] hasDriver(): device 0000:12:02.0 driver is iavf
I0921 15:42:38.095931 2991726 driver.go:83] BindDefaultDriver(): device 0000:12:02.0 already bound to default driver iavf
I0921 15:42:38.095941 2991726 utils.go:281] setNetdevMTU(): set MTU for device 0000:12:02.0 to 1500
I0921 15:42:38.096185 2991726 driver.go:79] BindDefaultDriver(): bind device 0000:12:02.1 to default driver
I0921 15:42:38.096202 2991726 driver.go:112] hasDriver(): device 0000:12:02.1 driver is iavf
I0921 15:42:38.096208 2991726 driver.go:83] BindDefaultDriver(): device 0000:12:02.1 already bound to default driver iavf
I0921 15:42:38.096223 2991726 utils.go:281] setNetdevMTU(): set MTU for device 0000:12:02.1 to 1500
W0921 15:42:38.096240 2991726 utils.go:290] setNetdevMTU(): fail to get interface name for 0000:12:02.1: GetNetName(): no net directory under pci device 0000:12:02.1: "lstat /sys/bus/pci/devices/0000:12:02.1/net: no such file or directory"
W0921 15:42:39.096347 2991726 utils.go:290] setNetdevMTU(): fail to get interface name for 0000:12:02.1: GetNetName(): no net directory under pci device 0000:12:02.1: "lstat /sys/bus/pci/devices/0000:12:02.1/net: no such file or directory"
```

Reply:
@SchSeba do you think it is due to we are not using a reliable way to check the readiness of VF device?

If we unbind/bind default driver when setting mtu, will it erase the admin mac address configured in previous step?
Can we guarantee that VF will no longer disappear after the second unbind/bind ?